AAU-P5-Moodle/moodle-1
# Add - Quiz Editor Menu - Adding Questions

# Description
NOTE: Please merge #137 before this PR 

This PR has been created to fix the issues in the styling_setup branch. It is now possible to create questions and add answers to the corresponding question. These are stored in the database when clicking "Add Question". The UI also shows a list of the questions present in the quiz. Lastly, we distinguish between a student and a lecturer, where only the lecturer can edit the quiz.

## What is added/changed/removed
**Quiz Editor Menu to**
- Create Questions
- Add answers
- Submit questions to DB
- Get overview of existing questions

## Issue/fixes reference
https://github.com/orgs/AAU-P5-Moodle/projects/2?pane=issue&itemId=82025880&issue=AAU-P5-Moodle%7Cmoodle-1%7C46, https://github.com/orgs/AAU-P5-Moodle/projects/2/views/1?pane=issue&itemId=85503814&issue=AAU-P5-Moodle%7Cmoodle-1%7C54, and https://github.com/orgs/AAU-P5-Moodle/projects/2/views/1?pane=issue&itemId=82021250&issue=AAU-P5-Moodle%7Cmoodle-1%7C45

## Testing
**Automatic Testing**
- Behat tests for the UI changes
- PHPunit tests for the external classes interacting with DB

**Manual Testing**
- Try adding questions and see that they appear both in the database and in the list in the quiz editor menu
- Login as a student and check the quiz editing menu does not appear for students

# Checks
- [x] Updated / added tests for these changes
